### PR TITLE
updates for appveyor & travis & reogranize

### PR DESCRIPTION
**Pull request by ttodua**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,16 +14,15 @@ before_install:
...
-- pip install requests pyopenssl
-- pip install --ignore-installed six
-- pip install tox twine
-- pip install aiohttp cryptography
+- git fetch --depth=1 --no-tags --verbose --progress
+- pip install tox cryptography requests # common packages
+- pip install twine # used to publish python package
+- pip install pyopenssl aiohttp # used to run real-time tests
 - php -i | grep php.ini
-- php --version
```
Comment (ttodua) [inline .travis.yml:26]: this was not needed, as above command `php -i` already exposes the version, i.e. see 1596th line: https://app.travis-ci.com/github/ccxt/ccxt/builds/263197729#L1595

File: .travis.yml [hunk 2]
```diff
@@ -14,16 +14,15 @@ before_install:
...
+# common commands
 - npm install
 - git checkout package.json package-lock.json
 - git config pull.rebase false
-- git fetch --depth=1 --tags --verbose --progress
-- pip install requests pyopenssl
-- pip install --ignore-installed six
-- pip install tox twine
-- pip install aiohttp cryptography
+- git fetch --depth=1 --no-tags --verbose --progress
```
Comment (ttodua) [inline .travis.yml:21]: we don't need --tags there, as it wastes log/time.

File: .travis.yml [hunk 3]
```diff
@@ -14,16 +14,15 @@ before_install:
...
+- pip install tox cryptography requests # common packages
+- pip install twine # used to publish python package
+- pip install pyopenssl aiohttp # used to run real-time tests
```
Comment (ttodua) [inline .travis.yml:24]: the commands are same, just they are sorted to be similar in `appveryor.yml` , thus more readable when looking into both `.yml` files.

File: appveyor.yml [hunk 4]
```diff
@@ -8,23 +8,16 @@ install:
   - set CI=true
   # - npm install --global npm@latest
   - set PATH=%APPDATA%\npm;C:\Python38;C:\Python38\Scripts;%PATH%
-  # Remove python2.7 from path
-  - set PATH=%PATH:C:\Python27;=%
-  - set PATH=%PATH:C:\Python27\Scripts;=%
-  - echo %PATH%
```
Comment (ttodua) [inline appveyor.yml:14]: we long ago dropped support for p2.7, no need to refer them

File: appveyor.yml [hunk 5]
```diff
@@ -8,23 +8,16 @@ install:
...
+  # common commands
   - npm install
-  - python --version
-  - pip --version
```
Comment (ttodua) [inline appveyor.yml:17]: moved these commands into `test_script:` block, where are other `--version` outputs.

File: appveyor.yml [hunk 6]
```diff
@@ -8,23 +8,16 @@ install:
...
   - python -m pip install --upgrade pip setuptools
-  - pip install tox
-  - pip install requests
-  - pip install cryptography
```
Comment (ttodua) [inline appveyor.yml:21]: these 3 commands are now inline (like we have in `travis.yml`)

File: build-appveyor.sh [hunk 7]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
```
Comment (ttodua) [inline build-appveyor.sh:1]: this is copy of `build.sh` (of travis), only removed "test" parts.

**Comment by ttodua** [comment]
@carlosmiei parallelly I have been trying to do similar for appveryor, but atm something doesn't word (namely `sh: build.sh` line).

**Comment by carlosmiei** [comment]
> @carlosmiei parallelly I have been trying to do similar for appveryor, but atm something doesn't word (namely `sh: build.sh` line).

@ttodua I can check it later but I don't think it's a priority right now since we managed to make travis very fast and that is the most important part.

**Comment by ttodua** [comment]
@carlosmiei the thing is that, even though it is not important (ofc I agree) there is a **need** that this test also passed to get confirmation about build success/failure. even though travis might succeed and publish package quickly, it needs additional 20+ mins till we get confirmation from appveyor (and in the meantime the orange dot is still blinking, so on first look we cant easily see whether build was fully ok for both platforms). so, even though not critical and priority, it would be quite much help if appveyor also did faster. 

lastly, it doesn't involve huge changes anymore, we just need to execute build.sh, and inside it I've added `if (is_travis)` logic to execute travis-only commands)

File: build.sh [hunk 8]
```diff
@@ -52,32 +54,51 @@ function run_tests {
...
+# temporarily remove the below scripts from diff
+if [ "$IS_TRAVIS" = "TRUE" ]; then
+  diff=$(git diff origin/master --name-only)
+  diff=$(echo "$diff" | sed -e "s/^build\.sh//")
+  diff=$(echo "$diff" | sed -e "s/^appveyor\.yml//")
+  diff=$(echo "$diff" | sed -e "s/^package\.json//")
+  diff=$(echo "$diff" | sed -e "s/python\/qa\.py//")
+  diff=$(echo "$diff" | sed -e "s/python\/tox\.ini//")
+else
+  git remote set-branches origin 'master'
```
Comment (ttodua) [inline build.sh:83]: in appveyor it seems we need to manually fetch master

**Pull request closed without merging**